lib/ts/stacks.ts

```diff
@@ -76,7 +76,66 @@ namespace Stacks {
 
         return Controller;
     }
+
     export function addController(name: string, controller: ControllerDefinition) {
         application.register(name, createController(controller));
-    };
+    }
+
+    export interface TooltipOptions {
+        placement: string;
+        classNames: string;
+    }
+
+    export function setTooltipHtml(element: Element, html: string, options?: TooltipOptions) {
+        setTooltip(element, popover => popover.innerHTML = html, options);
+    }
+
+    export function setTooltipText(element: Element, text: string, options?: TooltipOptions) {
+        setTooltip(element, popover => popover.textContent = text, options);
+    }
+
+    var generatedTooltipIdNumber = 0;
+
+    function setTooltip(element: Element, fn: (popover: HTMLElement) => void, options?: TooltipOptions) {
+
+        var controllerText = element.getAttribute("data-controller") || "";
+        if (!/(^|\s)s-tooltip(\s|$)/.test(controllerText)) {
```

Comment:
Is this for supporting space delimited controller strings? Use word boundaries instead:

`/\bs-tooltip\b/`

This will match all the following scenarios:

```
data-controller="s-tooltip"

data-controller="s-popover s-tooltip"

data-controller="s-popover s-tooltip s-something"

data-controller="s-tooltip s-popover"
```

Reply:
No can do: `/\bs-tooltip\b/.test('s-tooltip-foo') === true`